tests/trainer/test_trainer.py

```diff
@@ -1664,6 +1694,26 @@ def test_pytorch_profiler_nested(tmpdir):
         assert pa[n] == expected_[n]
 
 
+@pytest.mark.skipif(not _TORCH_GREATER_EQUAL_1_8, reason="Need at least PyTorch 1.8")
+def test_pytorch_profiler_trainer_new_api(tmpdir):
+    """Ensure that the profiler can be given to the training and default step are properly recorded. """
+
+    model = BoringModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        limit_train_batches=10,
+        limit_val_batches=10,
+        profiler=PyTorchProfiler(path_to_export_trace=tmpdir),
```

Comment:
Can you parametrize this test so we check both possible inputs?

`pytest.mark.parametrize('profiler', ('pytorch', PyTorchProfiler))`

Reply:
Done !